src/pfaffian.jl

```diff
@@ -87,28 +87,77 @@ function _pfaffian!(A::SkewHermTridiagonal{<:Real})
     return pf
 end
 
-pfaffian!(A::Union{SkewHermitian{<:Real},SkewHermTridiagonal{<:Real}})= _pfaffian!(A)
-pfaffian(A::Union{SkewHermitian{<:Real},SkewHermTridiagonal{<:Real}})= pfaffian!(copyeigtype(A))
+# Using the Parley-Reid algorithm from https://arxiv.org/abs/1102.3440 as implented
+# in https://github.com/KskAdch/TopologicalNumbers.jl/blob/42bda634d47aecb67cfcd495d97e0723b0e73a4f/src/pfaffian.jl#L332
+function _pfaffian!(A::AbstractMatrix{<:Complex})
+    n = size(A, 1)
+    isodd(n) && return zero(eltype(A))
+    tau = Array{eltype(A)}(undef, n - 2)
+    pf = one(eltype(A))
+    @inbounds for k in 1:2:n-1
+        tauk = @view tau[k:end]
+
+        # Pivot if neccessary
+        @views kp = k + argmax(abs.(A[k+1:end, k]))
```

Comment:
You could eliminate the array allocation by:
```jl
@views kp = k + argmax(abs, A[k+1:end, k])
```

Reply:
Unfortunately `argmax(f, array)` gives the _value_ in array that maximises `f`. What would work are `findmax(abs, array)[2]` or `argmax(Iterators.map(f, array))`, but both turn out to be a bit slower: 

```julia
julia> @benchmark @views argmax(abs.(A[2:end,3]))
BenchmarkTools.Trial: 10000 samples with 169 evaluations.
 Range (min … max):  585.609 ns …  13.256 μs  ┊ GC (min … max): 0.00% … 93.51%
 Time  (median):     612.728 ns               ┊ GC (median):    0.00%
 Time  (mean ± σ):   649.525 ns ± 463.065 ns  ┊ GC (mean ± σ):  3.13% ±  4.18%

  ▇▆▄▇█▅▄▇▆▃▂▃  ▁▂▁                      ▁                      ▂
  █████████████▆████▆▇▆▆▄▃▄▅▂▃▄▄▅▄▂▇▇█▇▅███▇▆▆█▇▆▅▃▆▆▆▅▅▅▅▃▅▄▄▄ █
  586 ns        Histogram: log(frequency) by time        962 ns <

 Memory estimate: 544 bytes, allocs estimate: 5.

julia> @benchmark @views argmax(Iterators.map(abs, A[2:end,3]))
BenchmarkTools.Trial: 10000 samples with 18 evaluations.
 Range (min … max):  933.944 ns … 164.654 μs  ┊ GC (min … max): 0.00% … 98.68%
 Time  (median):       1.013 μs               ┊ GC (median):    0.00%
 Time  (mean ± σ):     1.067 μs ±   1.647 μs  ┊ GC (mean ± σ):  1.52% ±  0.99%

      ▇█           ▁
  ▅▇▂▂██▄▃█▆▃▃▇▄▃▃▄█▃▂▂▄▅▂▂▂▃▃▂▂▂▂▃▂▂▂▁▂▃▂▂▂▂▂▂▃▂▂▂▁▂▂▃▂▂▂▂▂▂▁▂ ▃
  934 ns           Histogram: frequency by time         1.49 μs <

 Memory estimate: 240 bytes, allocs estimate: 5.

julia> @benchmark @views findmax(abs, A[2:end,3])[2]
BenchmarkTools.Trial: 10000 samples with 18 evaluations.
 Range (min … max):  967.778 ns … 157.201 μs  ┊ GC (min … max): 0.00% … 98.74%
 Time  (median):       1.014 μs               ┊ GC (median):    0.00%
 Time  (mean ± σ):     1.067 μs ±   1.565 μs  ┊ GC (mean ± σ):  1.46% ±  0.99%

  ▇█▄▂▅▆▃▁▂▆▅▁▁▃▆▄▁ ▄▅▂▁  ▂▁    ▂▁     ▁      ▂▁      ▁       ▁ ▂
  ██████████████████████▇▇██▇▆▅▅██▇▆▆▃██▇▆▃▆▃▇██▇▅▃▁▁███▅▅▅▁▁▅█ █
  968 ns        Histogram: log(frequency) by time       1.49 μs <

 Memory estimate: 304 bytes, allocs estimate: 7.

```

Above is with a random 50×50 matrix, but the ordering remains the same with a 1000×1000 matrix.

Reply:
If you interpolate `$A` into the benchmark, I find that the `Iterators.map` version is significantly faster:
```jl
julia> A = rand(100,100);

julia> using BenchmarkTools

julia> @btime @views argmax(abs.($A[2:end,3]));
  501.716 ns (1 allocation: 896 bytes)

julia> @btime @views argmax(Iterators.map(abs, $A[2:end,3]));
  119.438 ns (0 allocations: 0 bytes)
```

Reply:
Oops, the 101 of julia benchmarking. Another PR with this fixed is coming in a couple minutes.